### PR TITLE
auth on @searchable

### PR DESCRIPTION
**Pull request by SwaySway**
*Issue #, if available:*
re #1043
*Description of changes:*
added auth logic for `@searchable`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/graphql-auth-transformer/src/resources.ts [hunk 1]
```diff
@@ -1063,4 +1063,40 @@ identityClaim: "${rule.identityField || rule.identityClaim || DEFAULT_IDENTITY_F
...
+     * ES EXPRESSIONS
+     */
+
+    public makeESItemsExpression() {
+        // generate es expresion to appsync
+        return compoundExpression([
+            set(ref('es_items'), list([])),
+            forEach(
+                ref('entry'),
+                ref('context.result.hits.hits'),
```
Comment (nikhname) [inline packages/graphql-auth-transformer/src/resources.ts]: Is context.results.hits.hits intentional?

File: packages/graphql-transformers-e2e-tests/src/cognitoUtils.ts [hunk 2]
```diff
@@ -163,3 +176,212 @@ export async function createUserPoolClient(client: CognitoClient, userPoolId: st
...
+    const identityPoolRoleMap = new cfnCognito.IdentityPoolRoleAttachment({
+        IdentityPoolId: { Ref: 'IdentityPool' } as unknown as string,
+        Roles: {
+            'unauthenticated': { 'Fn::GetAtt': [ 'UnauthRole', 'Arn' ] },
+            'authenticated': { 'Fn::GetAtt': [ 'AuthRole', 'Arn' ] }
+        }
+    });
+
+    const userPoolClientWeb = new cfnCognito.UserPoolClient({
+        ClientName: USER_POOL_CLIENTWEB_NAME,
```
Comment (nikhname) [inline packages/graphql-transformers-e2e-tests/src/cognitoUtils.ts]: Why don't we generate a secret for userPoolClientWeb?

File: packages/graphql-auth-transformer/src/resources.ts (hunk 1 above)
Comment (SwaySway) [reply packages/graphql-auth-transformer/src/resources.ts]: Yup that's the response given back from elasticsearch
A hits object which contains list of hits.

File: packages/graphql-transformers-e2e-tests/src/cognitoUtils.ts (hunk 2 above)
Comment (SwaySway) [reply packages/graphql-transformers-e2e-tests/src/cognitoUtils.ts]: I don't think this is needed as the clientID is this e2e is testing auth resolver logic against Elasticsearch responses.

File: packages/graphql-auth-transformer/src/ModelAuthTransformer.ts [hunk 3]
```diff
@@ -1469,6 +1480,33 @@ All @auth directives used on field definitions are performed when the field is r
         }
     }
 
+    private protectSearchQuery(ctx: TransformerContext, def: ObjectTypeDefinitionNode,
+        resolverResourceId: string, rules: AuthRule[]) {
+            const resolver = ctx.getResource(resolverResourceId);
+            if (!rules || rules.length === 0 || !resolver) {
```
Comment (yuth) [inline packages/graphql-auth-transformer/src/ModelAuthTransformer.ts]: When woukd `resolver` be falsy? Should we not protect if resolver is `falsy`

Comment (SwaySway) [reply packages/graphql-auth-transformer/src/ModelAuthTransformer.ts]: The `resolver` would be falsy if say if the respective query for searchable is set to null.
```gql
@searchable(queries: null)
```

**Pull request merged**